### PR TITLE
Show exact balance user needs in error banner

### LINKED ISSUE #315: State the exact amount user should have in their account in the error message "Not enough funds to make transaction"
**What's important about this?**
Atm, when a user doesn't have enough funds to make a transaction the error message on the send screen, says just "Need more funds to send payment". 
So when the user has exactly 1$ and tries to send exactly 1$ they get confused why the transaction doesn't go through, since it says nothing that they need to cover tx fee as well.
The support team have received multiple request from users related to this issue.

**What's done?**
The error message says "You should have {exact amount} to make this transaction. Please add funds or try sending less."

**What's involved in doing this work**
Updating the error message

### PR DESCRIPTION
**Pull request by isabellewei**
### Description

Updated error banner shown when the user attempts to make a transaction that they don't have enough funds for to show the exact amount that is needed

### Tested

Screenshot_1622947044

### How others should test

Attempt to send a payment greater than the cUSD balance

### Related issues

- Fixes #315 

### Backwards compatibility

Yes

### CODE REVIEW

**Review by MuckT — COMMENTED** [review]
Can we make it so no fractional cents display?
| Min Transfer Amount | User Display |
| ----- | ----- |
| 55.000326408702 | 55.01 |
| 12.999 | 13 |
| 5.01 | 5.01 |

File: packages/mobile/src/send/SendAmount.tsx [hunk 1]
```diff
@@ -245,7 +247,11 @@ function SendAmount(props: Props) {
 
   const onSend = React.useCallback(() => {
     if (!isDollarBalanceSufficient) {
-      dispatch(showError(ErrorMessages.NSF_TO_SEND))
+      dispatch(
+        showError(ErrorMessages.NSF_TO_SEND, null, {
+          amountNeeded: minimumAmount,
+        })
+      )
```
Comment (jeanregisser) [inline packages/mobile/src/send/SendAmount.tsx:254]: I think we want to display the local amount and its symbol.
Similar to what we do for escrows
https://github.com/celo-org/wallet/blob/3aff7b630d899fec38fdb4fb6e2df0828a5bc053/packages/mobile/src/send/SendAmount.tsx#L200-L206

This makes me think we need a function equivalent of `<CurrencyDisplay />` so we can format money amounts as text only for cases like this.
No need to do this in this PR though ;)

File: packages/mobile/src/send/SendAmount.tsx [hunk 2]
```diff
@@ -181,6 +182,7 @@ function SendAmount(props: Props) {
 
   const isAmountValid = parsedLocalAmount.isGreaterThanOrEqualTo(DOLLAR_TRANSACTION_MIN_AMOUNT)
   const isDollarBalanceSufficient = isAmountValid && newAccountBalance.isGreaterThan(0)
+  const minimumAmount = roundUp(dollarAmount.plus(estimateFeeDollars || 0), 2)
```
Comment (jeanregisser) [inline packages/mobile/src/send/SendAmount.tsx:185]: `minimumAmount` needs to be added to the dependencies of `useCallback` where it's used.

File: packages/mobile/src/send/SendAmount.tsx [hunk 3]
```diff
@@ -275,7 +286,14 @@ function SendAmount(props: Props) {
...
       })
     }
-  }, [recipientVerificationStatus, addressValidationType, dollarAmount, getTransactionData, origin])
+  }, [
+    recipientVerificationStatus,
+    addressValidationType,
+    dollarAmount,
+    getTransactionData,
+    origin,
+    minimumAmount,
```
Comment (jeanregisser) [inline packages/mobile/src/send/SendAmount.tsx:295]:
```suggestion
    minimumAmount,
    localCurrencyExchangeRate,
    localCurrencySymbol,
```

**Review by jeanregisser — APPROVED** [review]
One last small adjustment, but already approved ;)

**Comment by Celoqa** [comment]
https://valora.testquality.com requirement [R37](https://valora.testquality.com/project/12128/testcase/286485/requirement/11406).

**Comment by Celoqa** [comment]
@isabellewei  @MuckT Need one clarification: 
- Is this changes done for only Send cUSD flow?  or we are planning to change banner for CELO as well. 
- " Not enough Celo Dollars to make exchange" banner is shown when user tried to "Buy" , "Sell" , "Withdraw CELO" so are we planning to add exact balance user needs in error banner  for CELO as well ?

**Comment by MuckT** [comment]
https://valora.testquality.com requirement [R58](https://valora.testquality.com/project/12128/testcase/286485/requirement/11552).

**Comment by Celoqa** [comment]
Hi @MuckTI have verified this issue on Latest android build V1.16.0 & Testflight build V1.16.0(66) and updated the status in test qulaity tool.
**Testquality tool link:**

**Android:**

- https://valora.testquality.com/project/12128/plan/20854/run/148956/result/2432215

**iOS:**

- https://valora.testquality.com/project/12128/plan/20854/run/148956/result/2432224

**Query ticket:**
- https://github.com/celo-org/wallet/issues/848

**Verified On Devices:** iPhone SE (14.2), Redmi Note 8 (9.0)
CC: @isabellewei

**Comment by Celoqa** [comment]
Hi @muckti have verified this issue on Latest android build V1.16.1 & Testflight build V1.16.1(67) and updated the status in test qulaity tool.
Testquality tool link:

Android:

https://valora.testquality.com/project/12128/plan/20854/run/148956/result/2432215
iOS:

https://valora.testquality.com/project/12128/plan/20854/run/148956/result/2432224
Query ticket:

- #848 
Verified On Devices: iPhone 6+ (12.4.5), Vivo v11 pro (10)
CC: @isabellewei

**Comment by Celoqa** [comment]
Hi @MuckT, We have verified this issue on Latest android build V1.16.0 in Android 11.0 and everything is working as expected.
**Device Details**: Google Pixel 4a(11.0)

**Pull request merged**